waku/waku_rln_relay/group_manager/on_chain/group_manager.nim

```diff
@@ -73,6 +73,8 @@ type
     # in event of a reorg. we store 5 in the buffer. Maybe need to revisit this,
     # because the average reorg depth is 1 to 2 blocks.
     validRootBuffer*: Deque[MerkleNode]
+    # this variable tracks the last seen head
+    lastSeenBlockHead*: BlockNumber
```

Comment:
right, I'll just move it to the newHeadCallback, thanks

Reply:
Addressed in 3c9d4bef

Reply:
> So the lastSeenBlockHead should be the last head block known by the blockchain.

I think its safer to get this from `eth_blockNumber` endpoint rather than relying on the newHeadcallback. If the eth node fails to keep you updated with the latest head (but without disconnecting) then your lastSeenBlockHead will be outdated.

But if you use `eth_blockNumber`, since its req/rep, you ensure that your latest head is correct. If then node fails to reply, then its not ready.

Reply:
Addressed in 15bfde44